Description: Add support for disabling gtk overlay scrollbars
Author: miketwebster@gmail.com
Origin: upstream, https://github.com/linuxmint/cinnamon-session/commit/96f25057cac202e13c338886d00bc0383394ad26
Bug: https://github.com/linuxmint/Cinnamon/issues/5306
Last-Update: 2016-05-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: cinnamon-session/cinnamon-session/main.c
===================================================================
--- cinnamon-session.orig/cinnamon-session/main.c	2016-05-16 13:41:10.916602897 +0200
+++ cinnamon-session/cinnamon-session/main.c	2016-05-18 22:26:58.891272339 +0200
@@ -306,11 +306,13 @@
         bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
         textdomain (GETTEXT_PACKAGE);
 
-        GSettings *settings = g_settings_new ("org.cinnamon.SessionManager");
+        GSettings *settings;
+
+        settings = g_settings_new ("org.cinnamon.SessionManager");
         if (g_settings_get_boolean (settings, "debug")) {
             debug = TRUE;
         }
-        g_object_unref(settings);
+        g_clear_object(&settings);
 
         sa.sa_handler = SIG_IGN;
         sa.sa_flags = 0;
@@ -384,6 +386,17 @@
         /* Make QT5 apps follow the GTK style */
         csm_util_setenv ("QT_STYLE_OVERRIDE", "gtk");
 
+        /* GTK Overlay scrollbars */
+        settings = g_settings_new ("org.cinnamon.desktop.interface");
+
+        if (g_settings_get_boolean (settings, "gtk-overlay-scrollbars")) {
+            csm_util_setenv ("GTK_OVERLAY_SCROLLING", "1");
+        } else {
+            csm_util_setenv ("GTK_OVERLAY_SCROLLING", "0");
+        }
+
+        g_clear_object (&settings);
+
         client_store = csm_store_new ();
 
         /* Talk to logind before acquiring a name, since it does synchronous
